### PR TITLE
fix(mobile): Rework topbar mobile layout

### LINKED ISSUE #1341: Video icon is clipped when in a chat
Issue: The users name and pfp is also covered and related
Screenshot:
image

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Rework the topbar buttons in chat for the mobile view. Fixes things like buttons being too crowded or going out of the screen.
- Buttons are now hidden behind a "More" button that when clicked will show those additional options
- Rework pinned message layout in mobile
- Rework edit group layout in mobile

### Which issue(s) this PR fixes 🔨

- Resolve #1341

### CODE REVIEW

**Comment by Flemmli97** [comment]
~~needs layout change~~

**Comment by XileHorizon** [comment]
Can we adjust the pins to be a bit narrower to fit better in the mobile layout?

Edit: I figured I could explain better just in case, I think the pins on narrow displays would look better if we decreased the width of the box to stay on the screen, stacked the buttons, and instead of continuing names to a new line when they exceed their width limit, just cut them off.
Before:

image
After:
mobile pins

**Comment by phillsatellite** [comment]
Along with what @XileHorizon said we should probably do the same for the Edit Groups 
Screenshot 2023-12-07 at 11 04 47 AM

Another thing I noticed is the call button should say "Call" when user has experimental flagged on 
Screenshot 2023-12-07 at 11 03 10 AM

**Comment by Flemmli97** [comment]
even though i dont think that pinned thing fits in this pr i still added it if thats ok.

**Comment by Flemmli97** [comment]
> Along with what @XileHorizon said we should probably do the same for the Edit Groups Screenshot 2023-12-07 at 11 04 47 AM
> 
> Another thing I noticed is the call button should say "Call" when user has experimental flagged on Screenshot 2023-12-07 at 11 03 10 AM

wondering @XileHorizon if you have a figma or some ideas for edit groups layout?

**Comment by phillsatellite** [comment]
> > Along with what @XileHorizon said we should probably do the same for the Edit Groups Screenshot 2023-12-07 at 11 04 47 AM
> > Another thing I noticed is the call button should say "Call" when user has experimental flagged on Screenshot 2023-12-07 at 11 03 10 AM
> 
> wondering @XileHorizon if you have a figma or some ideas for edit groups layout?

If the Groups issue is a little out of scope of the Pr and youd rather just fix the call button and handle the groups in a new ticket that works for me Just lemme know 👍 😄

File: ui/src/layouts/chats/presentation/chat/controls.rs [hunk 1]
```diff
@@ -167,64 +198,48 @@ pub fn get_controls(cx: Scope<ChatProps>) -> Element {
...
         Button {
             icon: Icon::PhoneArrowUpRight,
             disabled: !state.read().configuration.developer.experimental_features || *call_pending.current() || call_in_progress,
             aria_label: "Call".into(),
             appearance: Appearance::Secondary,
-            tooltip: cx.render(rsx!(Tooltip {
-                arrow_position: ArrowPosition::Top,
-                text: if !state.read().configuration.developer.experimental_features { get_local_text("uplink.coming-soon") } else { get_local_text("uplink.call") }
-            })),
+            text: text_builder("uplink.coming-soon"),
```
Comment (stavares843) [inline ui/src/layouts/chats/presentation/chat/controls.rs]: what phil meant is that this locale should say "Call" instead of "Coming soon"

**Comment by Flemmli97** [comment]
> > > Along with what @XileHorizon said we should probably do the same for the Edit Groups Screenshot 2023-12-07 at 11 04 47 AM
> > > Another thing I noticed is the call button should say "Call" when user has experimental flagged on Screenshot 2023-12-07 at 11 03 10 AM
> > 
> > 
> > wondering @XileHorizon if you have a figma or some ideas for edit groups layout?
> 
> If the Groups issue is a little out of scope of the Pr and youd rather just fix the call button and handle the groups in a new ticket that works for me Just lemme know 👍 😄

since i already did the pinned msg thing i would just do this in this pr too

**Review by phillsatellite — APPROVED** [review]

File: ui/src/layouts/chats/presentation/chat/controls.rs [hunk 2]
```diff
@@ -53,11 +54,18 @@ pub fn get_controls(cx: Scope<ChatProps>) -> Element {
...
     let active_call = state.read().ui.call_info.active_call();
     let call_in_progress = active_call.is_some(); // active_chat.map(|chat| chat.id) == active_call.map(|call| call.conversation_id);
 
     let show_pinned = use_state(cx, || false);
 
+    use_effect(cx, &minimal, |_| {
+        to_owned![show_more];
+        async move {
+            show_more.set(false);
+        }
```
Comment (sdwoodbury) [inline ui/src/layouts/chats/presentation/chat/controls.rs]: recommend this:

```
if show_more {
    show_more.set(false);
}
```

Comment (stavares843) [reply ui/src/layouts/chats/presentation/chat/controls.rs]: marking this as resolved as flem addressed it

**Review by sdwoodbury — APPROVED** [review]

**Review by stavares843 — COMMENTED** [review]
minor but the video call is not perceptible that is a button that is deactivated


desktop

Captura de ecrã 2023-12-11, às 22 58 45


vs mobile

Captura de ecrã 2023-12-11, às 22 58 37

fine with improving in another PR

**Pull request merged**